### ISSUE TITLE
[Issue]: DirectML Updates break non sub-quadratic optimziations.

### ISSUE THREAD

**Issue by TeutonJon78** [vladmandic/automatic#1711]
### Issue Description

After the recent DirectML updates, several of the optimziation methods that used to work no longer work and product OOM errors.

Specifically, on my RX580 8GB I was using doggettx and able to do 512x912 images.  After the DML update, that produces OOM messages.

cdc803b127e4aa660982799186d74b491165d15f -- everything worked as expected

6b4b863fd0ac645f407dcaaefd61a73c2da99b29 -- starting with this as the next working commit the exact same settings produce OOM errors for doggettx. sub-quadratic attention optimizations still work for the same image size.

I believe the error is in the modules/sd_hijack_optimziations.py function get_available_vram()

Old version:
```
   elif shared.device.type == 'privateuseone':
        mem_total, mem_active = torch.dml.memory_stats(shared.device)
        return mem_total - mem_active * (1 << 20)
```
print of torch.dml.memory_stats(shared.device) for:
Doggettx, 512x512: (8589934592, 7450)
Doggettx, 512x912: (8589934592, 7376)
Subquadratic, 512x912: (8589934592, 7393)

The function would return the amount of free memory.

New Version:
```
    elif shared.device.type == 'privateuseone':
        mem_free, mem_total = torch.dml.mem_get_info(shared.device)
        return mem_total - mem_free
```
print of torch.dml.mem_get_info(shared.device)
Doggettx, 512x512: (831520768, 8589934592)
Doggettx, 512x912: (831520768, 8589934592) before OOM
Subquadratic, 512x912: (759169024, 8589934592)

The function would now return the amount of used memory. It should probably just return `mem_free` not `mem_total-mem_free`

### Version Platform Description

Happens on a fresh install with no user extensions installed as well.

```
Using VENV: D:\Graphic Design\AI\vladmandic-SD.Next\venv
22:04:39-470170 INFO     Starting SD.Next
22:04:39-476172 INFO     Python 3.10.11 on Windows
22:04:39-530721 INFO     Version: e2b33b81 Sat Jul 15 22:40:00 2023 -0400
22:04:39-879931 INFO     Using DirectML Backend
22:04:40-071786 INFO     Enabled extensions-builtin: ['a1111-sd-webui-lycoris', 'clip-interrogator-ext', 'LDSR',
                         'Lora', 'multidiffusion-upscaler-for-automatic1111', 'ScuNET', 'sd-dynamic-thresholding',
                         'sd-extension-system-info', 'sd-webui-agent-scheduler', 'sd-webui-controlnet',
                         'stable-diffusion-webui-images-browser', 'stable-diffusion-webui-rembg', 'SwinIR']
22:04:40-080787 INFO     Enabled extensions: ['openOutpaint-webUI-extension', 'openpose-editor',
                         'sd-webui-3d-open-pose-editor', 'sd-webui-aspect-ratio-helper',
                         'sd-webui-infinite-image-browsing', 'sd-webui-model-converter', 'sd-webui-neutral-prompt',
                         'sd-webui-photopea-embed', 'sd-webui-regional-prompter', 'stable-diffusion-webui-two-shot',
                         'ultimate-upscale-for-automatic1111']
22:04:40-093790 INFO     Verifying requirements
22:04:40-116795 INFO     Verifying packages
22:04:40-119796 INFO     Verifying repositories
22:04:42-815846 INFO     Verifying submodules
22:05:22-357914 INFO     Extensions enabled: ['a1111-sd-webui-lycoris', 'clip-interrogator-ext', 'LDSR', 'Lora',
                         'multidiffusion-upscaler-for-automatic1111', 'ScuNET', 'sd-dynamic-thresholding',
                         'sd-extension-system-info', 'sd-webui-agent-scheduler', 'sd-webui-controlnet',
                         'stable-diffusion-webui-images-browser', 'stable-diffusion-webui-rembg', 'SwinIR',
                         'openOutpaint-webUI-extension', 'openpose-editor', 'sd-webui-3d-open-pose-editor',
                         'sd-webui-aspect-ratio-helper', 'sd-webui-infinite-image-browsing',
                         'sd-webui-model-converter', 'sd-webui-neutral-prompt', 'sd-webui-photopea-embed',
                         'sd-webui-regional-prompter', 'stable-diffusion-webui-two-shot',
                         'ultimate-upscale-for-automatic1111']
22:05:22-364481 INFO     Verifying packages
22:05:22-371491 INFO     Extension preload: 0.0s D:\Graphic Design\AI\vladmandic-SD.Next\extensions-builtin
22:05:22-374484 INFO     Extension preload: 0.0s D:\Graphic Design\AI\vladmandic-SD.Next\extensions
22:05:22-388487 INFO     Server arguments: ['--autolaunch', '--use-directml', '--use-cpu', 'interrogate', '--medvram']
No module 'xformers'. Proceeding without it.
22:05:29-903522 INFO     Pipeline: Backend.ORIGINAL
```

### Acknowledgements

- [X] I have read the above and searched for existing issues

**Comment by TeutonJon78** [comment]
Changing the function to return `mem_free` does restore functionality to doggettx at 512x912. Subquadratic still works as well.

The inverted return value might explain why some people were seeing faster generation times/higher resolutions. The different return value would alter the chunking calculations in sub_quad_attention. That might be something to investigate on what that changed.

**Comment by vladmandic** [comment]
cc @lshqqytiger

**Comment by lshqqytiger** [comment]
That's right. This is why I didn't expect any performance improvement. I mistakenly thought that I replaced with same formula. Should be fixed in 0f81cfc2133a47b19088ae0452d560972921b809.
